### PR TITLE
fix: emit config object

### PR DESCRIPTION
**Pull request by aodhgan**

### CODE REVIEW

**Comment by CjHare** [comment]
Does this address an issue i.e. https://github.com/windranger-io/windranger-treasury/issues/365#issuecomment-1193996171 ?

**Review by CjHare — COMMENTED** [review]
question: motivation for this change?

_Seems like busy work_ 🤔

File: test/event/staking/staking-factory-events.ts [hunk 1]
```diff
@@ -42,18 +47,18 @@ export function stakingPoolCreated(
...
-
     expect(args?.creator).is.not.undefined
 
-    expect(args?.rewardTokens).is.not.undefined
-    expect(args?.stakeToken).is.not.undefined
-    expect(args?.epochStartTimestamp).is.not.undefined
-    expect(args?.epochDuration).is.not.undefined
-    expect(args?.minimumContribution).is.not.undefined
-    expect(args?.rewardType).is.not.undefined
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access */
```
Comment (CjHare) [inline test/event/staking/staking-factory-events.ts:52]: question: is this comment needed?

_I'm curious why the `treasury` field is an unsafe member_

File: test/event/staking/staking-factory-events.ts [hunk 2]
```diff
@@ -10,13 +10,18 @@ type RewardToken = {
...
+type Config = {
+    treasury: string
     rewardTokens: RewardToken[]
     stakeToken: string
-    epochStartTimestamp: BigNumber
-    epochDuration: BigNumber
+    epochStartTimestamp: number
+    epochDuration: number
     minimumContribution: BigNumber
+    minTotalPoolStake: BigNumber
```
Comment (CjHare) [inline test/event/staking/staking-factory-events.ts:23]: @octavioamu max & min pool stakes are needed, correct?

Comment (octavioamu) [reply test/event/staking/staking-factory-events.ts:23]: yes I basically use for the UI all the data on the struct 
```
library StakingPoolLib {
    enum RewardType {
        NONE,
        FIXED,
        FLOATING
    }

    struct Reward {
        IERC20 tokens;
        uint256 maxAmount;
        uint256 ratio; // only initialized for fixed
    }

    struct Config {
        uint256 daoId;
        uint128 minTotalPoolStake;
        uint128 maxTotalPoolStake;
        uint128 minimumContribution;
        uint32 epochDuration;
        uint32 epochStartTimestamp;
        address treasury;
        IERC20 stakeToken;
        Reward[] rewardTokens;
        RewardType rewardType;
    }
}
```

File: test/event/staking/staking-factory-events.ts [hunk 3]
```diff
@@ -10,13 +10,18 @@ type RewardToken = {
...
     creator: string
+    config: Config
+}
+type Config = {
+    treasury: string
     rewardTokens: RewardToken[]
     stakeToken: string
-    epochStartTimestamp: BigNumber
-    epochDuration: BigNumber
+    epochStartTimestamp: number
```
Comment (CjHare) [inline test/event/staking/staking-factory-events.ts:20]: question: why the change to type number?

_There are now both `number` and `BigNumber` being used_

**Comment by octavioamu** [comment]
@aodhgan @CjHare can we get this merged asap please?

**Comment by CjHare** [comment]
> @aodhgan @CjHare can we get this merged asap please?

Awaiting @aodhgan replies to questions.

File: test/event/staking/staking-factory-events.ts (hunk 1 above)
Comment (aodhgan) [reply test/event/staking/staking-factory-events.ts:55]: I think its because its nested inside the args.config and the linter isnt smart enough to pick up on the optional chaining

File: test/event/staking/staking-factory-events.ts (hunk 3 above)
Comment (aodhgan) [reply test/event/staking/staking-factory-events.ts]: These are now both BigNumber's. I think it's simpler this way.

Comment (CjHare) [reply test/event/staking/staking-factory-events.ts]: This PR definitely is showing them as being added as `number`

File: test/event/staking/staking-factory-events.ts (hunk 1 above)
Comment (CjHare) [reply test/event/staking/staking-factory-events.ts:55]: Right, yes, I believe that may be one reason I assign to a local var to avoid the nesting 🤔 

...but if that is the cause, why are the other lines not also being flagged by the linter?
_They are also nested inside `args.config`_

Comment (aodhgan) [reply test/event/staking/staking-factory-events.ts:55]: updated to assign to a local variable

File: test/event/staking/staking-factory-events.ts (hunk 3 above)
Comment (aodhgan) [reply test/event/staking/staking-factory-events.ts]: I think somehow (?!) this was stale

**Comment by aodhgan** [comment]
@CjHare please re-review (button to request not working)

**Comment by CjHare** [comment]
> @CjHare please re-review (button to request not working)

That's because I've already been requested, being one of the pending reviewers  😉 
_(I receive an update whenever there's a new push to this PR)_

File: test/contracts/staking/staking-factory.test.ts [hunk 4]
```diff
@@ -37,10 +37,11 @@ import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers'
 // Wires up Waffle with Chai
 chai.use(solidity)
 
-const EPOCH_DURATION = 60
+const EPOCH_DURATION = BigNumber.from(60)
```
Comment (CjHare) [inline test/contracts/staking/staking-factory.test.ts:40]: suggestion: Use the short hand for `BigNumber` e.g. `60n`

This may be a personal style choice, but as I've used the shorthand everywhere in the Bonds, it might be a de-facto standard 🤷

Comment (aodhgan) [reply test/contracts/staking/staking-factory.test.ts:40]: Thats the shorthand for the javascript built-in`bigint` vs. ethers exported`BigNumber` though, right? Will I switch to using `bigint` everywhere in these tests?

Comment (CjHare) [reply test/contracts/staking/staking-factory.test.ts:40]: Does your code require `BigNumber` or `BigNumberish`?

Much of the TypeChain code seem to cater for `BigNumberish`, which allows using `bigint`, making the code look pretty nice (what you've got there with the casting looks a lot like Java from many years ago :wink)

**Review by CjHare — APPROVED** [review]
